src/main/java/com/excelsiorjet/api/tasks/ClasspathEntry.java

```diff
@@ -18,37 +18,154 @@
  *  along with Excelsior JET API.
  *  If not, see <http://www.gnu.org/licenses/>.
  *
-*/
+ */
+
 package com.excelsiorjet.api.tasks;
 
+import com.excelsiorjet.api.tasks.config.DependencySettings;
+import com.excelsiorjet.api.util.Utils;
+
 import java.io.File;
 
 /**
- * Java application classpath entry.
+ * Representation of Excelsior JET's project classpath entry.
+ *
+ * @author Aleksey Zhidkov
  */
 public class ClasspathEntry {
 
-    private final File file;
-    private final boolean isLib;
+    public final File path;
+
+    public final ProtectionType protect;
+
+    public final OptimizationType optimize;
+
+    public final PackType pack;
+
+    public final String packagePath;
+
+    public final Boolean disableCopyToPackage;
+
+    public final boolean isMainArtifact;
+
+    public ClasspathEntry(DependencySettings dependencySettings, boolean isMainArtifact) {
+        this(dependencySettings.path, deriveProtect(dependencySettings), deriveOptimize(dependencySettings), PackType.fromString(dependencySettings.pack), dependencySettings.packagePath, dependencySettings.disableCopyToPackage, isMainArtifact);
+    }
+
+    private static ProtectionType deriveProtect(DependencySettings dependencySettings) {
+        if (dependencySettings.protect == null && dependencySettings.isLibrary != null) {
+            return dependencySettings.isLibrary ? ProtectionType.NOT_REQUIRED : ProtectionType.ALL;
+        } else if (dependencySettings.protect != null) {
+            return ProtectionType.fromString(dependencySettings.protect);
+        } else {
+            return null;
+        }
+    }
+
+    private static OptimizationType deriveOptimize(DependencySettings dependencySettings) {
+        if (dependencySettings.optimize == null && dependencySettings.isLibrary != null) {
+            return dependencySettings.isLibrary ? OptimizationType.AUTO_DETECT : OptimizationType.ALL;
+        } else if (dependencySettings.optimize != null) {
+            return OptimizationType.fromString(dependencySettings.optimize);
+        } else {
+            return null;
+        }
+    }
+
+    public ClasspathEntry(File path, ProtectionType protect, OptimizationType optimize, PackType pack, String packagePath, Boolean disableCopyToPackage, boolean isMainArtifact) {
+        this.path = path;
+        this.protect = protect;
+        this.optimize = optimize;
+        this.pack = pack;
+        if (packagePath != null) {
+            this.packagePath = packagePath;
+        } else {
+            this.packagePath = isMainArtifact ? "" : null;
+        }
+        this.disableCopyToPackage = disableCopyToPackage;
+        this.isMainArtifact = isMainArtifact;
+    }
 
-    public ClasspathEntry(File file, boolean isLib) {
-        this.file = file;
-        this.isLib = isLib;
+    public ClasspathEntry isMainArtifact(boolean isMainArtifact) {
```

Comment:
this method is not used and other changes are also not needed except handling optimize/protect